pkg/datapath/linux/node_ids.go

```diff
@@ -187,6 +187,25 @@ func (n *linuxNodeHandler) unmapNodeID(ip string) error {
 	return nil
 }
 
+// diffAndUnmapNodeIPs takes two lists of node IP addresses: new and old ones.
+// It unmaps the node IP to node ID mapping for all the old IP addresses that
+// are not in the list of new IP addresses.
+func (n *linuxNodeHandler) diffAndUnmapNodeIPs(oldIPs, newIPs []nodeTypes.Address) {
+nextOldIP:
+	for _, oldAddr := range oldIPs {
+		for _, newAddr := range newIPs {
+			if newAddr.IP.String() == oldAddr.IP.String() {
```

Comment:
Any reason we are not using `net.IP.Equal()` here?

Reply:
None. I pushed an update with only rebase + that change.